### ISSUE TITLE
Error producing ndarray from parallel list comprehension

### ISSUE THREAD

**Issue by rborder** [numba/numba#4569]
Error producing ndarray from parallel list comprehension--works fine serially with ndarrays, works parallel when returning list, both break it:

## Minimal reproducer:

```
import numpy as np
from numba import njit, prange

@njit(parallel=True)
def get_lengths_as_list_par(xx): # works
    return [len(xx[i]) for i in prange(len(xx))]

@njit(parallel=False)
def get_lengths_as_ndarray_serial(xx): # works
    return np.array([len(xx[i]) for i in range(len(xx))])

@njit(parallel=True)
def get_lengths_as_ndarray_par(xx): # breaks
    return np.array([len(xx[i]) for i in prange(len(xx))])

xx = [np.array([1,2,3], dtype=int),np.array([1,2], dtype=int)]

get_lengths_as_list_par(xx) ## works fine
get_lengths_as_ndarray_serial(xx) ## works fine
get_lengths_as_ndarray_par(xx) ## produces below error
```
## Software versions

- numba: 0.42.1
- numpy: 1.16.2
- python: 3.6.8 

## Traceback:
```
KeyError                            Traceback (most recent call last)
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/errors.py in new_error_context(fmt_, *args, **kwargs)
    616     try:
--> 617         yield
    618     except NumbaError as e:

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_block(self, block)
    258                                    loc=self.loc, errcls_=defaulterrcls):
--> 259                 self.lower_inst(inst)
    260 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_inst(self, inst)
    307             ty = self.typeof(inst.target.name)
--> 308             val = self.lower_assign(ty, inst)
    309             self.storevar(val, inst.target.name)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_assign(self, ty, inst)
    453         elif isinstance(value, ir.Expr):
--> 454             return self.lower_expr(ty, value)
    455 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_expr(self, resty, expr)
    900         elif expr.op in ('getiter', 'iternext'):
--> 901             val = self.loadvar(expr.value.name)
    902             ty = self.typeof(expr.value.name)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in loadvar(self, name)
   1070         """
-> 1071         ptr = self.getvar(name)
   1072         return self.builder.load(ptr)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in getvar(self, name)
   1064         """
-> 1065         return self.varmap[name]
   1066 

KeyError: '$0.12'

During handling of the above exception, another exception occurred:

LoweringError                       Traceback (most recent call last)
<ipython-input-50-6d1e380882d3> in <module>()
     16 get_lengths_as_list_par(xx) ## works fine
     17 get_lengths_as_ndarray_serial(xx) ## works fine
---> 18 get_lengths_as_ndarray_par(xx) ## works fine

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/dispatcher.py in _compile_for_args(self, *args, **kws)
    365                     e.patch_message(''.join(e.args) + help_msg)
    366             # ignore the FULL_TRACEBACKS config, this needs reporting!
--> 367             raise e
    368 
    369     def inspect_llvm(self, signature=None):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/dispatcher.py in _compile_for_args(self, *args, **kws)
    322                 argtypes.append(self.typeof_pyval(a))
    323         try:
--> 324             return self.compile(tuple(argtypes))
    325         except errors.TypingError as e:
    326             # Intercept typing error that may be due to an argument

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler_lock.py in _acquire_compile_lock(*args, **kwargs)
     30         def _acquire_compile_lock(*args, **kwargs):
     31             with self:
---> 32                 return func(*args, **kwargs)
     33         return _acquire_compile_lock
     34 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/dispatcher.py in compile(self, sig)
    653 
    654             self._cache_misses[sig] += 1
--> 655             cres = self._compiler.compile(args, return_type)
    656             self.add_overload(cres)
    657             self._cache.save_overload(sig, cres)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/dispatcher.py in compile(self, args, return_type)
     80                                       args=args, return_type=return_type,
     81                                       flags=flags, locals=self.locals,
---> 82                                       pipeline_class=self.pipeline_class)
     83         # Check typing error if object mode is used
     84         if cres.typing_error is not None and not flags.enable_pyobject:

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in compile_extra(typingctx, targetctx, func, args, return_type, flags, locals, library, pipeline_class)
    924     pipeline = pipeline_class(typingctx, targetctx, library,
    925                               args, return_type, flags, locals)
--> 926     return pipeline.compile_extra(func)
    927 
    928 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in compile_extra(self, func)
    372         self.lifted = ()
    373         self.lifted_from = None
--> 374         return self._compile_bytecode()
    375 
    376     def compile_ir(self, func_ir, lifted=(), lifted_from=None):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in _compile_bytecode(self)
    855         """
    856         assert self.func_ir is None
--> 857         return self._compile_core()
    858 
    859     def _compile_ir(self):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in _compile_core(self)
    842         self.define_pipelines(pm)
    843         pm.finalize()
--> 844         res = pm.run(self.status)
    845         if res is not None:
    846             # Early pipeline completion

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler_lock.py in _acquire_compile_lock(*args, **kwargs)
     30         def _acquire_compile_lock(*args, **kwargs):
     31             with self:
---> 32                 return func(*args, **kwargs)
     33         return _acquire_compile_lock
     34 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in run(self, status)
    253                     # No more fallback pipelines?
    254                     if is_final_pipeline:
--> 255                         raise patched_exception
    256                     # Go to next fallback pipeline
    257                     else:

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in run(self, status)
    244                 try:
    245                     event(stage_name)
--> 246                     stage()
    247                 except _EarlyPipelineCompletion as e:
    248                     return e.result

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in stage_nopython_backend(self)
    715         """
    716         lowerfn = self.backend_nopython_mode
--> 717         self._backend(lowerfn, objectmode=False)
    718 
    719     def stage_compile_interp_mode(self):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in _backend(self, lowerfn, objectmode)
    664             self.library.enable_object_caching()
    665 
--> 666         lowered = lowerfn()
    667         signature = typing.signature(self.return_type, *self.args)
    668         self.cr = compile_result(

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in backend_nopython_mode(self)
    651                 self.calltypes,
    652                 self.flags,
--> 653                 self.metadata)
    654 
    655     def _backend(self, lowerfn, objectmode):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/compiler.py in native_lowering_stage(targetctx, library, interp, typemap, restype, calltypes, flags, metadata)
   1046         lower = lowering.Lower(targetctx, library, fndesc, interp,
   1047                                metadata=metadata)
-> 1048         lower.lower()
   1049         if not flags.no_cpython_wrapper:
   1050             lower.create_cpython_wrapper(flags.release_gil)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower(self)
    176         if self.generator_info is None:
    177             self.genlower = None
--> 178             self.lower_normal_function(self.fndesc)
    179         else:
    180             self.genlower = self.GeneratorLower(self)

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_normal_function(self, fndesc)
    217         # Init argument values
    218         self.extract_function_arguments()
--> 219         entry_block_tail = self.lower_function_body()
    220 
    221         # Close tail of entry block

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_function_body(self)
    242             bb = self.blkmap[offset]
    243             self.builder.position_at_end(bb)
--> 244             self.lower_block(block)
    245 
    246         self.post_lower()

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/lowering.py in lower_block(self, block)
    257             with new_error_context('lowering "{inst}" at {loc}', inst=inst,
    258                                    loc=self.loc, errcls_=defaulterrcls):
--> 259                 self.lower_inst(inst)
    260 
    261     def create_cpython_wrapper(self, release_gil=False):

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/contextlib.py in __exit__(self, type, value, traceback)
     97                 value = type()
     98             try:
---> 99                 self.gen.throw(type, value, traceback)
    100             except StopIteration as exc:
    101                 # Suppress StopIteration *unless* it's the same exception that

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/errors.py in new_error_context(fmt_, *args, **kwargs)
    623         from numba import config
    624         tb = sys.exc_info()[2] if config.FULL_TRACEBACKS else None
--> 625         six.reraise(type(newerr), newerr, tb)
    626 
    627 

/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/six.py in reraise(tp, value, tb)
    657         if value.__traceback__ is not tb:
    658             raise value.with_traceback(tb)
--> 659         raise value
    660 
    661 else:

LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
'$0.12'

File "<ipython-input-50-6d1e380882d3>", line 12:
def get_lengths_as_ndarray_par(xx):
    return np.array([len(xx[i]) for i in prange(len(xx))])
    ^

[1] During: lowering "$implicit0.3549.3583 = getiter(value=$0.12)" at <ipython-input-50-6d1e380882d3> (12)
-------------------------------------------------------------------------------
This should not have happened, a problem has occurred in Numba's internals.
```

**Comment by rborder** [comment]
This is easy to work around but probably shouldn't be happening :)

**Comment by stuartarchibald** [comment]
Thanks for the report. The IR produced is invalid, seems like the `parfor_pass` itself is the culprit.
```
label 0:
    xx = arg(0, name=xx)                     ['xx']
    $0.9 = global(len: <built-in function len>) ['$0.9']
    $0.11 = call $0.9(xx, func=$0.9, args=[Var(xx, issue4569.py (22))], kws=(), vararg=None) ['$0.11', '$0.9', 'xx']
    del $0.9                                 []
    $implicit0.117.153 = getiter(value=$0.12) ['$0.12', '$implicit0.117.153']
    del $0.12                                []
    index.139 = const(int, -1)               ['index.139']
    len_func.141 = global(range_iter_len: <intrinsic range_iter_len>) ['len_func.141']
    size.140 = call len_func.141($implicit0.117.153, func=len_func.141, args=(Var($implicit0.117.153, issue4569.py (22)),), kws=(), vararg=None) ['$implicit0.117.153', 'len_func.141', 'size.140']
    del len_func.141                         []
    del $implicit0.117.153                   []
    size_tuple.142 = build_tuple(items=[Var(size.140, issue4569.py (20))]) ['size.140', 'size_tuple.142']
    del size.140                             []
    empty_func.144 = global(unsafe_empty_inferred: <intrinsic empty_inferred>) ['empty_func.144']
    array.143 = call empty_func.144(size_tuple.142, func=empty_func.144, args=(Var(size_tuple.142, issue4569.py (20)),), kws=[], vararg=None) ['array.143', 'empty_func.144', 'size_tuple.142']
    del size_tuple.142                       []
    del empty_func.144                       []
---begin parfor 1---
index_var =  parfor_index.150
LoopNest(index_variable = parfor_index.150, range = (0, $0.11, 1))
init block:
label 81:
    $6.3.132 = xx                            ['$6.3.132', 'xx']
    $6.5.134 = getitem(value=xx, index=parfor_index.150) ['$6.5.134', 'parfor_index.150', 'xx']
    $6.5.134_shape.148 = getattr(value=$6.5.134, attr=shape) ['$6.5.134', '$6.5.134_shape.148']
    $6.5.134_size0.149 = static_getitem(value=$6.5.134_shape.148, index=0, index_var=None) ['$6.5.134_shape.148', '$6.5.134_size0.149']
    array.143[index.139] = $6.5.134_size0.149 ['$6.5.134_size0.149', 'array.143', 'index.139']
----end parfor 1----
    del xx                                   []
    del index.139                            []
    $0.16 = cast(value=array.143)            ['$0.16', 'array.143']
    del array.143                            []
    return $0.16                             ['$0.16']
```

**Comment by rborder** [comment]
Another problem that may or may not be the same bug:

Even the parallelized list version fails for large arguments:

```
@njit(parallel=False)
def get_lengths_serial(xx):
    return [np.int32(len(xx[i])) for i in range(len(xx))]

@njit(parallel=True)
def get_lengths_parallel(xx):
    return [np.int32(len(xx[i])) for i in prange(len(xx))]

t0 = get_lengths_parallel([np.array([1,2,3],dtype=np.int64),np.array([2,3],dtype=np.int64)]) ## works
t1 = get_lengths_serial([np.array([1,2,3],dtype=np.int64) for i in range(1000000)])                ## works
t2 = get_lengths_parallel([np.array([1,2,3],dtype=np.int64) for i in range(1000000)])             ## fails
```

Backtrace:

```
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x7ab54)[0x2b9959327b54]
/lib64/libc.so.6(+0x7ec18)[0x2b995932bc18]
/lib64/libc.so.6(realloc+0x1b2)[0x2b995932d752]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/runtime/_nrt_python.cpython-36m-x86_64-linux-gnu.so(+0x3d83)[0x2b9983581d83]
[0x2b995802710e]
[0x2b99581c40b8]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/tbbpool.cpython-36m-x86_64-linux-gnu.so(+0x45d1)[0x2b998696e5d1]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/tbbpool.cpython-36m-x86_64-linux-gnu.so(+0x4ee5)[0x2b998696eee5]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x29299)[0x2b9986ba0299]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x22a68)[0x2b9986b99a68]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x211d4)[0x2b9986b981d4]
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x1d33a)[0x2b9986b9433a]
/lib64/libpthread.so.0(+0x7e25)[0x2b9958787e25]
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
======= Backtrace: =========
/lib64/libc.so.6(clone+0x6d)[0x2b99593a534d]
/lib64/libc.so.6(+0x7ab54)[0x2b9959327b54]
======= Memory map: ========
/lib64/libc.so.6(+0x7ec18)[0x2b995932bc18]
/lib64/libc.so.6(realloc+0x1b2)[0x2b995932d752]
======= Backtrace: =========
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/runtime/_nrt_python.cpython-36m-x86_64-linux-gnu.so(+0x3d83)[0x2b9983581d83]
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/lib64/libc.so.6(+0x7ab54)[0x2b9959327b54]
[0x2b995802710e]
2b9957ff9000-2b995801a000 r-xp 00000000 fd:03 8390604                    /usr/lib64/ld-2.17.so
2b995801a000-2b995801b000 rw-p 00000000 00:00 0 
2b995801b000-2b9958022000 r--s 00000000 fd:03 8412581                    /usr/lib64/gconv/gconv-modules.cache
2b9958022000-2b9958023000 rwxp 00000000 00:00 0 
2b9958023000-2b9958024000 rw-s 00000000 00:12 3342430                    /dev/shm/sem.5zByWi (deleted)
2b9958024000-2b9958028000 r-xp 00000000 00:00 0 
2b9958028000-2b99581ad000 rw-p 00000000 00:00 0 
2b99581ad000-2b99581b0000 r--p 00000000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581b0000-2b99581bd000 r-xp 00003000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581bd000-2b99581c0000 r--p 00010000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581c0000-2b99581c1000 r--p 00012000 00:34 5940187889                 /projects/ribo7412/software/anaconda*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/lib64/libc.so.6(realloc+0x1b2/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/tbbpool.cpython-36m-x86_64-linux-gnu.so(+0x4ee5)[0x2b998696eee5]
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/runtime/_nrt_python.cpython-36m-x86_64-linux-gnu.so(+0x3d83)[0x2b9983581d83]
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x211d4)[0x2b9986b981d4]
[0x2b995802710e]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x1d33a)[0x2b9986b9433a]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/tbbpool.cpython-36m-x86_64-linux-gnu.so(+0x/lib64/libpthread.so.0(+0x*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
*** Error in `/projects/ribo7412/software/anaconda/envs/fwd/bin/python': realloc(): invalid next size: 0x000055cd98ee9060 ***
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x1d0c4)[0x2b9986b940c4]
/projects/ribo7412/software/anaconda/envs/fwd/lib/python3.6/site-packages/numba/npyufunc/../../../../libtbb.so.2(+0x1d33a)[0x2b9986b9433a]
/lib64/libpthread.so.0(+0x7e25)[0x2b9958787e25]
/lib64/libc.so.6(clone+0x6d)[0x2b99593a534d]
======= Memory map: ========
/envs/fwd/lib/libgcc_s.so.1
2b99581c1000-2b99581c2000 rw-p 00013000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581c2000-2b99581c3000 rw-p 00000000 00:00 0 
2b99581c3000-2b99581cb000 r-xp 00000000 00:00 0 
2b99581cb000-2b99581d2000 rw-p 00000000 00:00 0 
2b99581d2000-2b9958212000 rw-p 00000000 00:00 0 
2b995821a000-2b995821b000 r--p 00021000 fd:03 8390604                    /usr/lib64/ld-2.17.so
2b995821b000-2b995821c000 rw-p 00022000 fd:03 8390604                    /usr/lib64/ld-2.17.so
2b995821c000-2b995821d000 rw-p 00000000 00:00 0 
2b995821d000-2b99584e5000 r-xp 00000000 00:34 5939809692                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libpython3.6m.so.1.0
2b99584e5000-2b99586e5000 ---p 002c8000 00:34 5939809692                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libpython3.6m.so.1.0
2b99586e5000-2b99586eb000 r--p 002c8000 00:34 5939809692                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libpython3.6m2b9957ff9000-2b995801a000 r-xp 00000000 fd:03 8390604                    /usr/lib64/ld-2.17.so
2b995801a000-2b995801b000 rw-p 00000000 00:00 0 
2b995801b000-2b9958022000 r--s 00000000 fd:03 8412581                    /usr/lib64/gconv/gconv-modules.cache
2b9958022000-2b9958023000 rwxp 00000000 00:00 0 
2b9958023000-2b9958024000 rw-s 00000000 00:12 3342430                    /dev/shm/sem.5zByWi (deleted)
2b9958024000-2b9958028000 r-xp 00000000 00:00 0 
2b9958028000-2b99581ad000 rw-p 00000000 00:00 0 
2b99581ad000-2b99581b0000 r--p 00000000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581b0000-2b99581bd000 r-xp 00003000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581bd000-2b99581c0000 r--p 00010000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581c0000-2b99581c1000 r--p 00012000 00:34 5940187889                 /projects/ribo7412/software/anaconda2b9957ff9000-2b995801a000 r-xp 00000000 fd:03 8390604                    /usr/lib64/ld-2.17.so
2b995801a000-2b995801b000 rw-p 00000000 00:00 0 
2b995801b000-2b9958022000 r--s 00000000 fd:03 8412581                    /usr/lib64/gconv/gconv-modules.cache
2b9958022000-2b9958023000 rwxp 00000000 00:00 0 
2b9958023000-2b9958024000 rw-s 00000000 00:12 3342430                    /dev/shm/sem.5zByWi (deleted)
2b9958024000-2b9958028000 r-xp 00000000 00:00 0 
2b9958028000-2b99581ad000 rw-p 00000000 00:00 0 
2b99581ad000-2b99581b0000 r--p 00000000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581b0000-2b99581bd000 r-xp 00003000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581bd000-2b99581c0000 r--p 00010000 00:34 5940187889                 /projects/ribo7412/software/anaconda/envs/fwd/lib/libgcc_s.so.1
2b99581c0000-2b99581c1000 r--p 00012000 00:34 5940187889                 /projects/ribo7412/software/anacondaAborted
```

**Comment by stuartarchibald** [comment]
Thanks for the report, I suspect that this is a different bug so will put it in a new issue.

**Comment by stuartarchibald** [comment]
xref: https://github.com/numba/numba/issues/4579

**Comment by DrTodd13** [comment]
I have solved the issue with the invalid IR.  The parallel=True code was always removing the prange calls but in this case it can't do that because the IR is using the prange for something else.  So, I've turned off the explicit removal of prange but instead allow dead code elimination to get rid of it, for which I added prange to the list of functions with no side-effects.  So, in most cases it still goes away but not for this case where it needs to stick around.  

There's still an issue with the code though and I know what it is and I'm working on a solution.

@rborder @stuartarchibald

**Comment by stuartarchibald** [comment]
@DrTodd13 thanks for looking into this. RE the DCE, this seems like a good approach and I'm of the view that making the code transforms safe is most important. The later issue with inlining closures containing a prange, any idea on the LOE? If it's quite high perhaps we can just issue a performance warning saying the transform can't be done and then add a fix as time allows?

**Comment by stuartarchibald** [comment]
As of mainline for 0.55, the OP does not reproduce, closing assuming fixed by #4588.
